[11.0 P1] Blazor Web can use `BasePath` component instead of `<base href="">` (#36428)
dotnet/AspNetCore.Docs: aspnetcore/blazor/host-and-deploy/app-base-path.md

If you're rendering a Blazor app from different documents (for example, `/Admin/B/C/` and `/Admin/D/E/`), you must take the app base path into account, or the base path is different when the app renders in each document and the resources are fetched from the wrong URLs.

Choose one of the following strategies to keep relative links resolving correctly:

> * Map resources dynamically using the document they were rendered on as the root. This requires custom routing logic (for example, <xref:Microsoft.AspNetCore.Routing.IDynamicEndpointMetadata> or a <xref:Microsoft.AspNetCore.Routing.MatcherPolicy>) and is typically reserved for highly dynamic sites.

Reviewer: We just need a blank line before the version moniker ...

```suggestion
* Map resources dynamically using the document they were rendered on as the root. This requires custom routing logic (for example, <xref:Microsoft.AspNetCore.Routing.IDynamicEndpointMetadata> or a <xref:Microsoft.AspNetCore.Routing.MatcherPolicy>) and is typically reserved for highly dynamic sites.

```

:::moniker range=">= aspnetcore-11.0"

> * Establish a consistent base path for the document and map all app resources beneath that path. Blazor Server/Auto apps can rely on the `BasePath` component to render `<base href>` automatically, while standalone WebAssembly apps (or scenarios that always use a fixed value) configure the literal `<base>` tag in the host page.

Reviewer: Gotcha ... then how about the following? ... as long as Blazor Server is kind'a out of the picture ...

```suggestion
* Establish a consistent base path for the document and map all app resources beneath that path. Blazor Web Apps can rely on the `BasePath` component to render `<base href>` automatically, while standalone WebAssembly apps (or scenarios that always use a fixed value) configure the literal `<base>` tag in the host page.
```

:::moniker-end

:::moniker range="< aspnetcore-11.0"

> * Establish a consistent base path for the document and map all app resources beneath that path. Configure the literal `<base>` tag in the host page for both Blazor Server and WebAssembly apps (or scenarios that always use a fixed value).

Reviewer: "Blazor Server" is a bit vague per the naming conventions at >=8.0 after BWAs came out. Articles almost always adopt "server-side Blazor" language in these cases to cover both Blazor Server and BWAs. How about? ...

```suggestion
* Establish a consistent base path for the document and map all app resources beneath that path. Configure the literal `<base>` tag in the host page for both server-side Blazor apps and Blazor WebAssembly apps (or scenarios that always use a fixed value).
```

> In some hosting scenarios, such as GitHub Pages and IIS sub-apps, the app base path must be set to the server's relative URL path of the app.

* In a server-side Blazor app, use ***either*** of the following approaches:

> :::moniker range=">= aspnetcore-11.0"

Reviewer: Again ... just some spacing to add ...

```suggestion

:::moniker range=">= aspnetcore-11.0"

```

    ```

    **The trailing slash is required.**

> :::moniker-end

Reviewer: ... and here ...

```suggestion

:::moniker-end
```
